mesonbuild/interpreter/dependencyfallbacks.py

```diff
@@ -219,10 +220,15 @@ def _get_cached_dep(self, name: str, kwargs: TYPE_nkwargs) -> T.Optional[Depende
         else:
             info = [mlog.blue('(cached)')]
             cached_dep = self.coredata.deps[for_machine].get(identifier)
+            from_disk_cache = True
 
         if cached_dep:
             found_vers = cached_dep.get_version()
             if not self._check_version(wanted_vers, found_vers):
+                if from_disk_cache:
+                    # We cached this dependency on disk from a previous run,
+                    # but it could got updated on the system in the meantime.
+                    return None
```

Comment:
I don't think we even need this flag, right? we're already inside `cached_dep`, so we basically just need to check if `override.explicit`, and error for that case, otherwise return None, or did I miss something?

Reply:
It could be written as `if not override:` but it felt better to have a proper variable name to describe that condition.

Reply:
Isn't that what comments are for?

Reply:
I'm concerned about having two variables to represent the same state, it's really easy for them to get out of sync and end up with subtly broken code.

Reply:
ok, updated :)